src/table.js

```diff
@@ -538,15 +543,99 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
+  const numberDefault = defaultValue === null ? NaN : defaultValue;
+  switch (type) {
+    case "string":
+      return value === "string"
+        ? value.trim()
+        : value || value === 0
+        ? value.toString()
+        : defaultValue;
+    case "boolean":
+      return value === true || value === "true"
```

Comment:
Do we want to add `value.trim() === "true"`, so that the string `"true "` that includes whitespace is also coerced to true?